### ISSUE TITLE
Use view and click tracking hooks on top stories component

### ISSUE THREAD

**Issue by jroebu14** [bbc/simorgh#9082]
**Is your feature request related to a problem? Please describe.**
As a ws product manager I want to know how many users who see the top stories promos on our pages click on the links so that I have a baseline to measure improvements to the promo .

**Describe the solution you'd like**
Enable tracking to the top stories block that will send a single view event when the block is in view and a single click event if any link within the block is clicked. The content of the view and click event must be the same for the block so ATI can calculate the click/view value.  The tracking must contain details of what service is being tracked.

Configure the Top Stories block on cps articles to enable block level tracking. This will give a view in ATI where we can track the views/clicks of the block.

Attach the view event to the first item in the top stories component and a click event on each link within.

e.g. https://www.bbc.com/pidgin/world-56829018

![image-2021-04-26-12-41-45-825](https://user-images.githubusercontent.com/4798332/116541714-bb660500-a8e3-11eb-8cf3-f168731858fd.png)

### The click event
Send a block level click event if any link/promo within the block is clicked. (we are only tracking the interaction and not tracking the link location)

Don't send a click event if any other part of the block is clicked.

`atc=PUB-[Campaign]-[Creation]-[Variant]-[Format]-[General Placement]-[Detailed Placement]-[Advertiser ID]-[Url]&type=AT`

```
(atc)
atc=PUB-[article-sty]-[top-stories]-[]-[]-[pidgin.world.story.56829018.page]-[]-[]-[]&type=AT
```

### The view/impression event

`ati=PUB-[Campaign]-[Creation]-[Variant]-[Format]-[General]-[Detailed]-[Advertiser]-[Url]&type=AT`

```
(ati)
ati=PUB-[article-sty]-[top-stories]-[]-[]-[pidgin.world.story.56829018.page]-[]-[]-[]&type=AT
```

### Example values from click/impression events

|                    |                                                                  |
| ------------------ | ---------------------------------------------------------------- |
| Campaign           | 'article-sty' (this is dynamic and relates to the page type)                                                        |
| Creation           | 'top-stories'                                                    |
| Variant            | empty                                                            |
| Format             | empty                                                            |
| General Placement  | _page_ eg pidgin.world.story.56829018.page (same as the p value) |
| Detailed Placement | empty                                                            |
| Advertiser ID      | empty                                                            |
| Url                | empty                                                            |

### ATI Tag Inspector
In the Tag Inspector these values should be present:
- Site - 598342
- Level2 - 70
- (Under -Ad)
- Campaign ID - article-sty
- Creation ID - top-stories
- General placement - pidgin.world.story.56829018.page

### Accessibility
The solution should have no adverse affects with any of our support assistive technology. This will require a developer accessiblity swarm to check that nothing unexpected is read out, such as 'clickable'. e.g. Only the content and HTML semantics should be announced for elements, nothing extra to the defined screen reader UX for components. Note as we are using the same technical solution, this testing should be carried out on either this Issue or https://github.com/bbc/simorgh/issues/9081 - which ever is completed first.

**Describe alternatives you've considered**

**Testing notes**
[Tester to complete]

Dev insight: Will Cypress tests be required or are unit tests sufficient? Will there be any potential regression? etc

- [x] This feature is expected to need manual testing.

**Additional context**
The WS event tracking matrix contains all tracking campaigns and which components are appear on which page types NEWSWORLDSERVICE/WS+event+matrix.

**Comment by JonBeeb** [comment]
Updated the campaign/creation names.

**Comment by JonBeeb** [comment]
Just checking that the ati and atc fields need to be the same 
current
ati=PUB-[sty]-[topStories]-[]-[]-[news::pidgin.news.story.51745682.page]-[]-[]-[]	
atc=PUB-[sty]-[topStories]-[]-[]-[news::pidgin.news.story.51745682.page]-[]-[]-[http://localhost:7080/pidgin/tori-51745682]

Expected
ati=PUB-[sty]-[topStories]-[]-[]-[news::pidgin.news.story.51745682.page]-[]-[]-[]	
atc=PUB-[sty]-[topStories]-[]-[]-[news::pidgin.news.story.51745682.page]-[]-[]-[]

**Comment by JonBeeb** [comment]
Hello, I've done some (final) tweaking on the naming, so it follows formatting of other stuff in ATI so now with -  [article-sty]-[top-stories]. If you could apply the change then hit some pages (preferably pidgin) I can check they appear in test ATI.

**Comment by paruchurisilpa** [comment]
Checked this on test using the following URL's:

* https://www.test.bbc.com/pidgin/tori-23146434
* https://www.test.bbc.com/hindi/23068699
* https://www.test.bbc.com/telugu/23145894


ATI Tag Inspector

In the Tag Inspector these values should be present:

    Site - 598342
    Level2 - 70
    (Under -Ad)
    Campaign ID - article-sty
    Creation ID - top-stories
    General placement - pidgin.world.story.56829018.page

* Clocking on 2nd/3rd story is making extra request
* Clicking on browser back button is making a request
* In ATI Tag Inspector I dont see Campaign ID, Creation ID

@jroebu14 I have noticed few things I need to clarify with you, will check tomorrow..

**Comment by paruchurisilpa** [comment]
Caught up with Jonathan and all the above looks good to me.

Click and Events are tracked correctly and no other requests are fired when you interact with any other elements on the page. Looks good to me.
